### PR TITLE
Add config option: authorization_failure_path

### PR DESCRIPTION
**Pull request by mcmire**
When a user is logging into Spree for the first time, they will get an
screen that asks if they want to authorize a connection to the OAuth
application.

If the user rejects this authorization, then by default, they will be
redirected to the Spree login page. The problem is, wu-unit overrides
this route so that the user is automatically sent to Gatekeeper instead.
So for wu-unit, we need a way to change the "authorization denied" path
to the root path (otherwise the user ends up right back on the
authorization screen).

This commit adds a configuration option called
`authorization_failure_path` that makes this possible. wu-unit will
eventually use it like this:

``` ruby
SpreeSocial.configure do |config|
  config.authorization_failure_path = lambda_or_string
end
```

### CODE REVIEW

File: lib/spree_social/configuration.rb [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
   class Configuration
     attr_accessor :find_authentication_method
+    attr_reader :authorization_failure_path
 
     def initialize
       @find_authentication_method = lambda do
         Spree::AuthenticationMethod.find_by!(environment: ::Rails.env)
       end
+
+      @authorization_failure_path = -> (context) { context.spree.login_path }
```
Comment (dfreerksen) [inline lib/spree_social/configuration.rb]: Should the default be set to `root_path` in case the configuration isn't set?

Comment (mcmire) [inline lib/spree_social/configuration.rb]: Maybe? I was going off of what happens in wu-unit by default, but perhaps `root_path` makes more sense. In fact, my plan was for wu-unit to use this option to set the path to the root path. In that case, maybe we don't even need this config option?

Comment (dfreerksen) [inline lib/spree_social/configuration.rb]: Yeah. Let's just redirect them to `root_path` instead of providing a configuration.

**Comment by mcmire** [comment]
This was merged as a373dfecbe51f1cd32669233d14daa3de9e8e611.

**Pull request closed without merging**